### PR TITLE
strengthen typing of is_java_array_tag

### PR DESCRIPTION
**Pull request by kroening**
- [X] Each commit message has a non-empty body, explaining why the change was made.
- [X] My contribution is formatted in line with CODING_STANDARD.md.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [X] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: fa29ee7b35be67abf2fb7820ba71ae35dd9aabc9).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/86365144
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

File: jbmc/src/java_bytecode/java_types.cpp [hunk 1]
```diff
@@ -828,14 +826,16 @@ void get_dependencies_from_generic_parameters_rec(
...
-    const irep_idt class_name(symbol_type.get_identifier());
-    if(is_java_array_tag(class_name))
+    if(is_java_array_tag(symbol_type))
     {
       get_dependencies_from_generic_parameters(
         java_array_element_type(symbol_type), refs);
     }
     else
+    {
+      const irep_idt class_name(symbol_type.get_identifier());
```
Comment (tautschnig) [inline jbmc/src/java_bytecode/java_types.cpp]: Nit pick: `const irep_idt &class_name`

Comment (kroening) [reply jbmc/src/java_bytecode/java_types.cpp]: ok

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 5d5c1eec6aab5b82e8a956609a276f1375e0fefe).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/86367476
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

File: jbmc/src/java_bytecode/java_types.cpp [hunk 2]
```diff
@@ -171,9 +169,9 @@ bool is_multidim_java_array_type(const typet &type)
 /// \param tag Tag of a struct
 /// \return True if the given string is a Java array tag, i.e., has a prefix
 /// of java::array[
-bool is_java_array_tag(const irep_idt& tag)
+bool is_java_array_tag(const symbol_typet &tag)
```
Comment (thk123) [inline jbmc/src/java_bytecode/java_types.cpp:172]: How does this strengthen the check? I think if you want to improve this, what we've been meaning to do for a while is introduce a `java_array_object_typet` (name up for discussion) that encapsulates all of these `java_array` methods a bit.

**Review by thk123 — COMMENTED** [review]
This needs a real TG bump

**Comment by TGWDB** [comment]
Closing due to age (no further comment on PR content), please reopen with rebase on develop if you intent to continue this work.

**Pull request closed without merging**